compiler/qsc/src/interpret.rs

```diff
@@ -482,6 +549,32 @@ impl Interpreter {
         self.run_with_sim(&mut sim, receiver, expr)
     }
 
+    /// Invokes the given callable with the given arguments using the current environment, simlator, and compilation.
```

Comment:
Have you given any thought into making this behave like `run`, in that it runs on a brand new simulator instance?

After all it's more or less intended to be a more ergonomic way of calling `run`... right?

Reply:
Right now that could work, yes, but if we eventually want to support returning more complicated types across the boundary into Python (like callables, lambdas with captures, or even qubits) they have to correlate back to the environment and simulator that is being kept alive. So in that sense, this is more like `eval_fragments` than it is like `run`. I'll move the function up to be near those to support that association and make it easier on our future selves if/when we need that.